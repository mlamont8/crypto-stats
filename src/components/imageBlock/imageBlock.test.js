import React from 'react';
import renderer from 'react-test-renderer';
import ImageBlock from './imageBlock';
import crypto from '../../images/crypto.png';

describe('ImageBlock component', () => {
    it('renders correctly', () => {
        const imageProps = { coinFrom: "ARK", coinName: "Test Coin Name", img: "https://url.com/testurl.jpg" };
        const renderedImage = renderer.create(
            <ImageBlock
                coinFrom={imageProps.coinFrom}
                coinName={imageProps.coinName}
                img={imageProps.img}
            />
        );
        expect(renderedImage.toJSON()).toMatchSnapshot();
    })
})