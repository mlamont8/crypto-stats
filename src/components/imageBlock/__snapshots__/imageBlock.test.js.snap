// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageBlock component renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    modalToggle={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt={undefined}
          className="imageIcon"
          data-src={undefined}
          height="40px"
          uk-img="true"
          width="40px"
/>,
        <span
          className="nameBlk"
>
           
          <FontAwesomeIcon
                    border={false}
                    className=""
                    fixedWidth={false}
                    flip={null}
                    icon={
                              Object {
                                        "icon": Array [
                                          448,
                                          512,
                                          Array [],
                                          "f061",
                                          "M190.5 66.9l22.2-22.2c9.4-9.4 24.6-9.4 33.9 0L441 239c9.4 9.4 9.4 24.6 0 33.9L246.6 467.3c-9.4 9.4-24.6 9.4-33.9 0l-22.2-22.2c-9.5-9.5-9.3-25 .4-34.3L311.4 296H24c-13.3 0-24-10.7-24-24v-32c0-13.3 10.7-24 24-24h287.4L190.9 101.2c-9.8-9.3-10-24.8-.4-34.3z",
                                        ],
                                        "iconName": "arrow-right",
                                        "prefix": "fas",
                                      }
                    }
                    inverse={false}
                    listItem={false}
                    mask={null}
                    pull={null}
                    pulse={false}
                    rotation={null}
                    size={null}
                    spin={false}
                    symbol={false}
                    transform={null}
          />
           
</span>,
        <div
          className="ibContainer-sr"
>
          <span>
                    <h2 />
          </span>
</div>,
        <div
          className="ibContainer-tr"
>
          <button
                    onClick={[Function]}
                    type="button"
          >
                    New Search
          </button>
</div>,
      ],
      "className": "infoBlock imageBlock",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": undefined,
          "className": "imageIcon",
          "data-src": undefined,
          "height": "40px",
          "uk-img": "true",
          "width": "40px",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            undefined,
            " ",
            <FontAwesomeIcon
              border={false}
              className=""
              fixedWidth={false}
              flip={null}
              icon={
                            Object {
                                          "icon": Array [
                                            448,
                                            512,
                                            Array [],
                                            "f061",
                                            "M190.5 66.9l22.2-22.2c9.4-9.4 24.6-9.4 33.9 0L441 239c9.4 9.4 9.4 24.6 0 33.9L246.6 467.3c-9.4 9.4-24.6 9.4-33.9 0l-22.2-22.2c-9.5-9.5-9.3-25 .4-34.3L311.4 296H24c-13.3 0-24-10.7-24-24v-32c0-13.3 10.7-24 24-24h287.4L190.9 101.2c-9.8-9.3-10-24.8-.4-34.3z",
                                          ],
                                          "iconName": "arrow-right",
                                          "prefix": "fas",
                                        }
              }
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              symbol={false}
              transform={null}
/>,
            " ",
            undefined,
          ],
          "className": "nameBlk",
        },
        "ref": null,
        "rendered": Array [
          undefined,
          " ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "border": false,
              "className": "",
              "fixedWidth": false,
              "flip": null,
              "icon": Object {
                "icon": Array [
                  448,
                  512,
                  Array [],
                  "f061",
                  "M190.5 66.9l22.2-22.2c9.4-9.4 24.6-9.4 33.9 0L441 239c9.4 9.4 9.4 24.6 0 33.9L246.6 467.3c-9.4 9.4-24.6 9.4-33.9 0l-22.2-22.2c-9.5-9.5-9.3-25 .4-34.3L311.4 296H24c-13.3 0-24-10.7-24-24v-32c0-13.3 10.7-24 24-24h287.4L190.9 101.2c-9.8-9.3-10-24.8-.4-34.3z",
                ],
                "iconName": "arrow-right",
                "prefix": "fas",
              },
              "inverse": false,
              "listItem": false,
              "mask": null,
              "pull": null,
              "pulse": false,
              "rotation": null,
              "size": null,
              "spin": false,
              "symbol": false,
              "transform": null,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          " ",
          undefined,
        ],
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <span>
            <h2 />
</span>,
          "className": "ibContainer-sr",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <h2 />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "h2",
          },
          "type": "span",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <button
            onClick={[Function]}
            type="button"
>
            New Search
</button>,
          "className": "ibContainer-tr",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "New Search",
            "onClick": [Function],
            "type": "button",
          },
          "ref": null,
          "rendered": "New Search",
          "type": "button",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt={undefined}
            className="imageIcon"
            data-src={undefined}
            height="40px"
            uk-img="true"
            width="40px"
/>,
          <span
            className="nameBlk"
>
             
            <FontAwesomeIcon
                        border={false}
                        className=""
                        fixedWidth={false}
                        flip={null}
                        icon={
                                    Object {
                                                "icon": Array [
                                                  448,
                                                  512,
                                                  Array [],
                                                  "f061",
                                                  "M190.5 66.9l22.2-22.2c9.4-9.4 24.6-9.4 33.9 0L441 239c9.4 9.4 9.4 24.6 0 33.9L246.6 467.3c-9.4 9.4-24.6 9.4-33.9 0l-22.2-22.2c-9.5-9.5-9.3-25 .4-34.3L311.4 296H24c-13.3 0-24-10.7-24-24v-32c0-13.3 10.7-24 24-24h287.4L190.9 101.2c-9.8-9.3-10-24.8-.4-34.3z",
                                                ],
                                                "iconName": "arrow-right",
                                                "prefix": "fas",
                                              }
                        }
                        inverse={false}
                        listItem={false}
                        mask={null}
                        pull={null}
                        pulse={false}
                        rotation={null}
                        size={null}
                        spin={false}
                        symbol={false}
                        transform={null}
            />
             
</span>,
          <div
            className="ibContainer-sr"
>
            <span>
                        <h2 />
            </span>
</div>,
          <div
            className="ibContainer-tr"
>
            <button
                        onClick={[Function]}
                        type="button"
            >
                        New Search
            </button>
</div>,
        ],
        "className": "infoBlock imageBlock",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": undefined,
            "className": "imageIcon",
            "data-src": undefined,
            "height": "40px",
            "uk-img": "true",
            "width": "40px",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              undefined,
              " ",
              <FontAwesomeIcon
                border={false}
                className=""
                fixedWidth={false}
                flip={null}
                icon={
                                Object {
                                                "icon": Array [
                                                  448,
                                                  512,
                                                  Array [],
                                                  "f061",
                                                  "M190.5 66.9l22.2-22.2c9.4-9.4 24.6-9.4 33.9 0L441 239c9.4 9.4 9.4 24.6 0 33.9L246.6 467.3c-9.4 9.4-24.6 9.4-33.9 0l-22.2-22.2c-9.5-9.5-9.3-25 .4-34.3L311.4 296H24c-13.3 0-24-10.7-24-24v-32c0-13.3 10.7-24 24-24h287.4L190.9 101.2c-9.8-9.3-10-24.8-.4-34.3z",
                                                ],
                                                "iconName": "arrow-right",
                                                "prefix": "fas",
                                              }
                }
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size={null}
                spin={false}
                symbol={false}
                transform={null}
/>,
              " ",
              undefined,
            ],
            "className": "nameBlk",
          },
          "ref": null,
          "rendered": Array [
            undefined,
            " ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "border": false,
                "className": "",
                "fixedWidth": false,
                "flip": null,
                "icon": Object {
                  "icon": Array [
                    448,
                    512,
                    Array [],
                    "f061",
                    "M190.5 66.9l22.2-22.2c9.4-9.4 24.6-9.4 33.9 0L441 239c9.4 9.4 9.4 24.6 0 33.9L246.6 467.3c-9.4 9.4-24.6 9.4-33.9 0l-22.2-22.2c-9.5-9.5-9.3-25 .4-34.3L311.4 296H24c-13.3 0-24-10.7-24-24v-32c0-13.3 10.7-24 24-24h287.4L190.9 101.2c-9.8-9.3-10-24.8-.4-34.3z",
                  ],
                  "iconName": "arrow-right",
                  "prefix": "fas",
                },
                "inverse": false,
                "listItem": false,
                "mask": null,
                "pull": null,
                "pulse": false,
                "rotation": null,
                "size": null,
                "spin": false,
                "symbol": false,
                "transform": null,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            " ",
            undefined,
          ],
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <span>
              <h2 />
</span>,
            "className": "ibContainer-sr",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <h2 />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "h2",
            },
            "type": "span",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <button
              onClick={[Function]}
              type="button"
>
              New Search
</button>,
            "className": "ibContainer-tr",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "New Search",
              "onClick": [Function],
              "type": "button",
            },
            "ref": null,
            "rendered": "New Search",
            "type": "button",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
